public/js/user/home.js

```diff
@@ -44,6 +51,12 @@ registerEmail.addEventListener('submit', (e) => {
   formData.forEach((value, key) => {
     dataregisterEmail[key] = value;
   });
+  const { email } = dataregisterEmail;
+  if (/^[\w.-_%+]+@[\w.-]+\.[a-zA-Z]{2,4}$/.test(email) === false) {
```

Comment:
@aminalakhsham You could pull out this email regex test into its own function, since it's being used twice.  Then you could update it by changing just one line, instead of two.

And of course if you end up needing email validation in another page too, you could then move it into its own file.

Reply:
i'll do that :)